quaint/src/connector/queryable.rs

```diff
@@ -101,13 +101,30 @@ pub trait Queryable: Send + Sync {
 
 /// A thing that can start a new transaction.
 #[async_trait]
-pub trait TransactionCapable: Queryable
-where
-    Self: Sized,
-{
+pub trait TransactionCapable: Queryable {
     /// Starts a new transaction
-    async fn start_transaction(&self, isolation: Option<IsolationLevel>) -> crate::Result<Transaction<'_>> {
-        let opts = TransactionOptions::new(isolation, self.requires_isolation_first());
-        Transaction::new(self, self.begin_statement(), opts).await
-    }
+    async fn start_transaction<'a>(
+        &'a self,
+        isolation: Option<IsolationLevel>,
+    ) -> crate::Result<Box<dyn Transaction + 'a>>;
+}
+
+macro_rules! impl_default_TransactionCapable {
```

Comment:
This is the only notable change to the Rust part since the last time it got reviewed: I had to remove default trait implementation and replace it with macro instead. Reason for that: default implementation requires `Self: Sized` where the trait singature itself does not. `Self:Sized` makes non-object-safe. We need to use `dyn TransactionCapabale` in JS connector wrapper in sql connctor now: https://github.com/prisma/prisma-engines/pull/4138/files#diff-fb223ab8a8714c0e8f66070ba712029ca38bad22f31b66fc04f30e73b68ef587R131